# Errors when running python3 $OPENRAM_HOME/openram.py myconfig

## Run python3 $OPENRAM_HOME/openram.py myconfig
### Error LOG
ERROR: file calibre.py: line 124: Unable to retrieve DRC results file. Is calibre set up?
Traceback (most recent call last):
 File “/mada/users/ywang646/OpenRAM/compiler/verify/calibre.py”, line 122, in run_drc
   f = open(drc_runset[‘drcSummaryFile’], “r”)
FileNotFoundError: [Errno 2] No such file or directory: ‘/tmp/openram_ywang646_15539_temp/dff_inv_2.drc.summary’

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
 File “/soe/ywang646/OpenRAM/compiler/openram.py”, line 61, in <module>
   name=OPTS.output_name)
 File “/mada/users/ywang646/OpenRAM/compiler/sram.py”, line 43, in __init__
   self.s.create_netlist()
 File “/mada/users/ywang646/OpenRAM/compiler/sram_1bank.py”, line 26, in create_netlist
   sram_base.create_netlist(self)
 File “/mada/users/ywang646/OpenRAM/compiler/sram_base.py”, line 72, in create_netlist
   self.add_modules()
 File “/mada/users/ywang646/OpenRAM/compiler/sram_base.py”, line 233, in add_modules
   self.control_logic = self.mod_control_logic(num_rows=self.num_rows)
 File “/soe/ywang646/OpenRAM/compiler/modules/control_logic.py”, line 28, in __init__
   self.create_netlist()
 File “/soe/ywang646/OpenRAM/compiler/modules/control_logic.py”, line 35, in create_netlist
   self.add_modules()
 File “/soe/ywang646/OpenRAM/compiler/modules/control_logic.py”, line 61, in add_modules
   dff = dff_inv()
 File “/soe/ywang646/OpenRAM/compiler/modules/dff_inv.py”, line 30, in __init__
   self.create_layout()
 File “/soe/ywang646/OpenRAM/compiler/modules/dff_inv.py”, line 45, in create_layout
   self.DRC_LVS()
 File “/soe/ywang646/OpenRAM/compiler/base/hierarchy_design.py”, line 77, in DRC_LVS
   debug.check(verify.run_drc(self.name, tempgds) == 0,“DRC failed for {0}“.format(self.name))
 File “/mada/users/ywang646/OpenRAM/compiler/verify/calibre.py”, line 124, in run_drc
   debug.error(“Unable to retrieve DRC results file. Is calibre set up?“,1)
 File “/mada/users/ywang646/OpenRAM/compiler/debug.py”, line 23, in error
   assert return_value==0
AssertionError


### Step1
Git pull PrivateRAM
### Step2
Copy /mada/software/setup.sh" to /.bashrc
### Step3
Add OpenRAM enviorment variables:

export OPENRAM_HOME="$HOME/PrivateRAM/compiler"

export OPENRAM_TECH="$HOME/PrivateRAM/technology"

export PYTHONPATH="$PYTHONPATH:$OPENRAM_HOME"

export FREEPDK45="/mada/software/techfiles/FreePDK45"

### Step4
Restart the shell
